src/core/meshcop/tcat_agent.cpp

```diff
@@ -501,6 +505,93 @@ Error TcatAgent::HandleStartThreadInterface(void)
     return error;
 }
 
+void SeralizeTcatAdvertisementTlv(uint8_t                 *aBuffer,
+                                  uint16_t                &aOffset,
+                                  TcatAdvertisementTlvType aType,
+                                  uint16_t                 aLength,
+                                  const uint8_t           *aValue)
+{
+    aBuffer[aOffset++] = (uint8_t)(aType << 4 | (aLength & 0xf));
+    memcpy(aBuffer + aOffset, aValue, aLength);
+    aOffset += aLength;
+}
+
+uint8_t *TcatAgent::GetAdvertisementData(uint16_t &aLen)
+{
+    aLen = 0;
+
+    if (mVendorInfo == nullptr)
+    {
+        return nullptr;
+    }
+
+    *reinterpret_cast<uint16_t *>(mAdvertisement) = LittleEndian::HostSwap16(OT_TOBLE_SERVICE_UUID);
```

Comment:
Can use `LittleEndian::WriteUint16()` here instead? (safer than casting and easier to read)

Reply:
fixed.